File: src/main/java/leetcode/Problem1503.java
package leetcode;

/**
 * https://leetcode.com/problems/last-moment-before-all-ants-fall-out-of-a-plank/
 */
public class Problem1503 {
    public int getLastMoment(int n, int[] left, int[] right) {
        // TODO
        return 0;
    }

    public static void main(String[] args) {
        Problem1503 prob = new Problem1503();
        System.out.println(prob.getLastMoment(4, new int[]{4,3}, new int[]{0,1})); // 4
        System.out.println(prob.getLastMoment(7, new int[]{}, new int[]{0,1,2,3,4,5,6,7})); // 7
        System.out.println(prob.getLastMoment(7, new int[]{0,1,2,3,4,5,6,7}, new int[]{})); // 7
        System.out.println(prob.getLastMoment(9, new int[]{5}, new int[]{4})); // 5
        System.out.println(prob.getLastMoment(6, new int[]{6}, new int[]{0})); // 6
    }
}

File: src/main/kotlin/leetcode/Problem1503.kt
package leetcode

import kotlin.math.max

/**
 * https://leetcode.com/problems/last-moment-before-all-ants-fall-out-of-a-plank/
 */
class Problem1503 {
    fun getLastMoment(n: Int, left: IntArray, right: IntArray): Int {
        var answer = 0
        for (i in left) {
           answer = max(answer, i)
        }
        for (i in right) {
            answer = max(answer, n - i)
        }
        return answer
    }
}